aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1673,24 +1716,41 @@ Tensor sparse_compressed_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize
                 "[blocksize=", blocksize,"] is not implemented.");
     return sparse_compressed_clone(self, blocksize, "to_sparse_bsc");
   }
-  AT_ERROR(
-      "sparse_compressed_to_sparse_bsc expected SparseBsr or SparseBsc layout but got ",
-      self.layout());
-  return self;
-}
-
-Tensor sparse_compressed_to_sparse_csc(const Tensor& self, c10::optional<int64_t> dense_dim_opt) {
-  if (dense_dim_opt.has_value()) {
-    AT_ERROR("sparse_compressed_to_sparse_csc conversion does not support specifying number of dense dimensions");
-  }
-  if (self.layout() == kSparseCsr) {
-    return sparse_compressed_to_flipped(self, c10::nullopt, "to_sparse_csc");
-  }
   if (self.layout() == kSparseCsc) {
-    return sparse_compressed_clone(self, c10::nullopt, "to_sparse_csc");
+    TORCH_CHECK(self.dim() == 2 + self.dense_dim(),
```

Comment:
```suggestion
    auto batch_ndim = sparse_csr::numBatchDimensions(self);
    TORCH_CHECK(batch_ndim > 0,
```

Reply:
Changed, but after some additional refactoring, in the next PR in the stack.